mex/common/models/schema.py

```diff
@@ -0,0 +1,139 @@
+from typing import Any, Optional, get_origin
```

Comment:
i still find the name `schema.py` a bit confusing. because the models actually describe the **filter** and **mapping** rules. they are only converted to a _schema_ once we call `model_json_schema()` on them (which we don't do in mex-common).

so what do you think about putting GenericRule, GenericField, generate_mapping_schema_for_mex_class, MAPPING_MODEL_BY_EXTRACTED_CLASS_NAME in a `mex.common.models.mapping` and the filter-model equivalent stuff in `mex.common.models.filter`?

Reply:
yes I agree. Looks much better now